internal/operators/odf/validations.go

```diff
@@ -168,16 +168,19 @@ func (o *operator) setStatusInsufficientResources(odfClusterResources *odfCluste
 // count all disks of drive type ssd or hdd
 func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID string, mode odfDeploymentMode) (int64, error) {
 	var countDisks int64
-	var err error
+	var smallDisks int64
 
 	for _, disk := range disks {
 		if (disk.DriveType == models.DriveTypeSSD || disk.DriveType == models.DriveTypeHDD) && installationDiskID != disk.ID && disk.SizeBytes != 0 {
-			if disk.SizeBytes < conversions.GbToBytes(o.config.ODFMinDiskSizeGB) {
-				err = fmt.Errorf("Insufficient resources to deploy ODF in %s mode. ODF requires a minimum of 3 hosts. Each host must have at least 1 additional disk of %d GB minimum and an installation disk.", strings.ToLower(string(mode)), o.config.ODFMinDiskSizeGB)
-			} else {
+			if disk.SizeBytes >= conversions.GbToBytes(o.config.ODFMinDiskSizeGB) {
 				countDisks++
+			} else {
+				smallDisks++
 			}
 		}
 	}
-	return countDisks, err
+	if countDisks == 0 && smallDisks != 0 {
+		return countDisks, fmt.Errorf("Insufficient resources to deploy ODF in %s mode. ODF requires a minimum of 3 hosts. Each host must have at least 1 additional disk of %d GB minimum and an installation disk.", strings.ToLower(string(mode)), o.config.ODFMinDiskSizeGB)
```

Comment:
isn't this message already duplicated [here](https://github.com/openshift/assisted-service/blob/890e9c7f8f1d178bf08c6f8c817c3f7ce776b39e/internal/operators/odf/validations.go#L162)?
in that case maybe this function can just count disks without validating the amount and raising an error

Reply:
it is, this becuse there is some test's  / validation  will be broken if we dont add it. 
or we will need to re-factor a lot of test's.

as I metion on top, after changing this function logic it will be esier to handle and create better validation flow.